File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/mutate/dao/RecordMutateDao.java
package ru.citeck.ecos.records3.record.op.mutate.dao;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.records3.record.op.atts.dto.RecordAtts;
import ru.citeck.ecos.records2.RecordRef;
import ru.citeck.ecos.records3.record.dao.RecordsDao;

public interface RecordMutateDao extends RecordsDao {

    @NotNull
    RecordRef mutate(@NotNull RecordAtts record);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/mutate/dao/RecordMutateDao.kt
package ru.citeck.ecos.records3.record.op.mutate.dao

import ru.citeck.ecos.records3.record.dao.RecordsDao
import ru.citeck.ecos.records3.record.op.atts.dto.LocalRecordAtts

interface RecordMutateDao : RecordsDao {

    fun mutate(record: LocalRecordAtts): String
}
